### DISCUSSION TITLE
Admin token expiration

### DISCUSSION THREAD

**Discussion by BPilot253** [pocketbase/pocketbase#1778] [Q&A] [answered]
Hello, I have this code currently that runs on start-up of my web server, however, just yesterday I tried to get some data from the database and it returned 403 Only admins can perform this action, since I guess it's been 14+ days since the last server restart.

```js
globals.database = new PocketBase("http://localhost:8090")
globals.database.admins.authWithPassword(process.env.DATABASE_EMAIL, process.env.DATABASE_PASSWORD)
```
Is there a way to make the admin token not expire at all? I know I can just check for every request if the token is valid but that isn't that ideal for me.

**Comment by ganigeorgiev** [comment, marked as answer]
"API keys/Permanent tokens" are currently not supported but it is in my todo and eventually will be added sometime in the future (_you can also check the discussion in https://github.com/pocketbase/pocketbase/discussions/342_).

You can change the duration of the generated admin token from the "Admin UI > Settings > Token Options > Admins auth token duration" field (max ~2 years).

Alternatively, you can also implement a timer that will call periodically `admins.authRefresh()` (_from security standpoint when combined with a short token duration this is slightly better_).

**Reply by BPilot253** [reply]
Alright, thank you, I think I'll go for the first option for the sake of simplicity, at least for now.
